src/components/shared/Entity/DetailsForm.tsx

```diff
@@ -57,40 +62,69 @@ function DetailsForm({ connectorTags, messagePrefix }: Props) {
         }
     }, [connectorID, setDetails]);
 
+    const accessGrantsOneOf = useMemo(() => {
```

Comment:
We need to "post process" some data so throwing those into memos. Keeping them apart since they can change independently.